### PR TITLE
Auto deploy doxygen to openvdb.org

### PR DESCRIPTION
**Pull request by Idclip**
Includes various doxygen fixes

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -49,17 +49,39 @@ jobs:
...
         github.event_name == 'workflow_dispatch' &&
         github.event.inputs.deploy == 'true'
       uses: peaceiris/actions-gh-pages@v3
       with:
-        github_token: ${{ secrets.GITHUB_TOKEN }}
+        deploy_key: ${{ secrets.ACTIONS_DEPLOY_KEY }}
         publish_dir: /usr/local/share/doc/OpenVDB/html
+        destination_dir: documentation/doxygen
+        external_repository: AcademySoftwareFoundation/openvdb-website
+        publish_branch: doxygen
```
Comment (Idclip) [inline .github/workflows/docs.yml:83]: This could also be the master branch to avoid having to do another PR we we allowed direct pushing to master

**Review by danrbailey — APPROVED** [review]
